vendor_old/sigs.k8s.io/controller-reconciler/pkg/genericreconciler/genericreconciler.go

```diff
@@ -68,7 +70,7 @@ func (gr *Reconciler) ReconcileResource(namespacedname types.NamespacedName) (re
 	period := DefaultReconcilePeriod
 	expected := []reconciler.Object{}
 	resource := gr.resource.(runtime.Object).DeepCopyObject()
-	nname := reflect.TypeOf(resource).String() + "/" + namespacedname.String()
+	nname := strings.Trim(reflect.TypeOf(resource).String(), "*") + "/" + namespacedname.String()
```

Comment:
What changed to require this string trimming?

Reply:
The line previously resulted in a name like `*v1alpha1.CDAPMaster`. The leading `*`  is present as TypeOf(resource) is a pointer. The `*` doesn't cause any unexpected behaviour presently as it's only used in log statements below. 
New k8s versions may not understand resource names with a leading `*`. Similar issue was fixed in https://github.com/cdapio/cdap-operator/pull/81